python/ray/data/_internal/logical/operators/read_operator.py

```diff
@@ -11,8 +11,17 @@ def __init__(
         self,
         datasource: Datasource,
         read_tasks: List[ReadTask],
+        estimated_num_blocks: int,
         ray_remote_args: Optional[Dict[str, Any]] = None,
     ):
-        super().__init__(f"Read{datasource.get_name()}", None, ray_remote_args)
+        if len(read_tasks) == estimated_num_blocks:
+            suffix = ""
+        else:
+            suffix = f"->SplitBlocks({int(estimated_num_blocks / len(read_tasks))})"
+        super().__init__(f"Read{datasource.get_name()}{suffix}", None, ray_remote_args)
         self._datasource = datasource
+        self._estimated_num_blocks = estimated_num_blocks
         self._read_tasks = read_tasks
+
+    def fusable(self) -> bool:
```

Comment:
I think the splitting logic can be implemented in the Map physical operator, instead of the read task. So we can still fuse read + map tasks, then just split the output of the map operator.

Reply:
The problem is that then we can't fan out to multiple processes, so you'd lose the "real" parallelism gained from the splitting.

Reply:
by "can't fan out to multiple processes", do you mean that for "read->map", map's parallelism must be the same as read? 
I checked the code, I think it's true for task based maps, but not for actor based maps. 
However, I think even for task based maps, we should also allow fan out between sequential ops. Because the parallelism of each stage can be configured based on how heavy each stage is.